aws/alarms/lambda/notify_slack/notify_slack.ts

```diff
@@ -0,0 +1,159 @@
+  const warningMessages = ["warning", "failure"];
+  const alarm_ok_status = '"newstatevalue":"ok"'; // This is the string that is returned when the alarm is reset
+
+  message = message.toLowerCase();
+
+  if (message.indexOf("sev1") != -1) return "SEV1";
+
+  for (var errorMessagesItem in errorMessages) {
+    if (
+      message.indexOf(errorMessages[errorMessagesItem]) != -1 &&
+      message.indexOf(alarm_ok_status) == -1 // is not an OK status
+    ) {
+      return "error";
+    } else if (message.indexOf(alarm_ok_status) != -1) {
+      return "alarm_reset";
+    }
+  }
+
+  for (var warningMessagesItem in warningMessages) {
+    if (
+      message.indexOf(warningMessages[warningMessagesItem]) != -1 &&
+      message.indexOf(alarm_ok_status) == -1 // is not an OK status
+    ) {
+      return "warning";
+    } else if (message.indexOf(alarm_ok_status) != -1) {
+      return "alarm_reset";
+    }
+  }
+
+  return "info";
+};
+
+export const handleCloudWatchLogEvent = async (event: any) => {
+  console.log("Received CloudWatch logs event: ", JSON.stringify(event));
+  var payload = Buffer.from(event.awslogs.data as string, "base64");
+
+  return new Promise<void>(function (resolve, reject) {
+    gunzip(payload, function (error, result) {
+      if (error) {
+        reject(error);
+      } else {
+        const parsedResult = JSON.parse(result.toString());
+
+        // We can get events with a `CONTROL_MESSAGE` type. It happens when CloudWatch checks if the Lambda is reachable.
+        if (parsedResult.messageType === "CONTROL_MESSAGE") return;
+
+        for (const log of parsedResult.logEvents) {
+          const logMessage = safeParseLogIncludingJSON(log.message);
+          // If logMessage is false, then the message is not JSON
+          if (logMessage) {
+            const message = `
+              ${logMessage.msg}
+              ${logMessage.error ? "\n".concat(logMessage.error) : ""}
+              ${logMessage.severity ? "\n\nSeverity level: ".concat(logMessage.severity) : ""}
+              `;
+            sendToSlack(parsedResult.logGroup, message, logMessage.level);
+            sendToOpsGenie(parsedResult.logGroup, message, logMessage.severity);
+            console.log(
+              JSON.stringify({
+                msg: `Event Data for ${parsedResult.logGroup}: ${JSON.stringify(
+                  logMessage,
+                  null,
+                  2
+                )}`,
+              })
+            );
+          } else {
+            // These are unhandled errors from the GCForms app only
+            sendToSlack(parsedResult.logGroup, log.message, "error");
+
+            console.log(
+              JSON.stringify({
+                msg: `Event Data for ${parsedResult.logGroup}: ${log.message}`,
+              })
+            );
+          }
+        }
+        resolve();
+      }
+    });
+  });
+};
+
+export const handleSnsEventFromCloudWatchAlarm = (event: any) => {
```

Comment:
We can rename `event` to something like `message` and specify the `string` type.

When calling the function you will be able to do:
```
} else if (event?.Records?.[0]?.Sns?.Message) {
      handleSnsEventFromCloudWatchAlarm(event.Records[0].Sns.Message);
```
(reasoning based on the official documentation https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html)